Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -743,35 +743,36 @@ internal static async void SetMasterLeftBarButton(UIViewController containerCont
 				return;
 			}
 
-			EventHandler handler = (o, e) => masterDetailPage.IsPresented = !masterDetailPage.IsPresented;
-
-			bool shouldUseIcon = masterDetailPage.Master.Icon != null;
-			if (shouldUseIcon)
+			await masterDetailPage.Master.ApplyNativeImageAsync(Page.IconProperty, icon =>
 			{
-				try
+				if (icon != null)
 				{
-					var icon = await masterDetailPage.Master.Icon.GetNativeImageAsync();
-					containerController.NavigationItem.LeftBarButtonItem = new UIBarButtonItem(icon, UIBarButtonItemStyle.Plain, handler);
+					try
+					{
+						containerController.NavigationItem.LeftBarButtonItem = new UIBarButtonItem(icon, UIBarButtonItemStyle.Plain, OnItemTapped);
+					}
+					catch (Exception)
+					{
+						// Throws Exception otherwise would catch more specific exception type
+					}
 				}
-				catch (Exception)
+
+				if(icon == null || containerController.NavigationItem.LeftBarButtonItem == null)
 				{
-					// Throws Exception otherwise would catch more specific exception type
-					shouldUseIcon = false;
+					containerController.NavigationItem.LeftBarButtonItem = new UIBarButtonItem(masterDetailPage.Master.Title, UIBarButtonItemStyle.Plain, OnItemTapped);
 				}
-			}
 
-			if (!shouldUseIcon)
```

Comment:
@mattleibow 
I moved all of this code up into the callback

https://github.com/xamarin/Xamarin.Forms/pull/4915/commits/e8d23bf746fcfb6c009c166618f7cee0571ad6ce

Otherwise it was just never setting the title if the icon wasn't specified